http_requests.py

```diff
@@ -107,6 +107,9 @@ def _request(self,
             del kwargs['headers']
         if 'data' in kwargs:
             headers['content-type'] = 'application/x-yaml'
+            if 'encoding' in kwargs:
```

Comment:
I think I asked this before: why not just support passing content-type via headers?

Reply:
Done